codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RustSettings.kt

```diff
@@ -81,7 +82,8 @@ class RustSettings(
     val runtimeConfig: RuntimeConfig,
     val codegenConfig: CodegenConfig,
     val license: String?,
-    val examplesUri: String? = null
+    val examplesUri: String? = null,
+    val customizationConfig: ObjectNode? = null
```

Comment:
I don't _love_ this name although I don't have a better one in mind. Maybe "pluginConfig"? Not sure. I just feel like `Customization` is sort of an internally used word that might not have clear meaning otherwise

Reply:
I didn't want to do `pluginConfig` since smithy-rs itself is a plugin for Smithy, so it's more like `pluginOfAPluginConfig` if we go that route. I'm definitely open to suggestions as I'm struggling to think of a better name.